image/image.go

```diff
@@ -102,6 +102,36 @@ func localSnaps(opts *Options) (*localInfos, error) {
 			info.Revision = snap.R(-1)
 			nameToPath[info.Name()] = snapName
 			local[snapName] = info
+
+			// crude way to find the .assert file
+			assertFile := strings.Replace(snapName, ".snap", ".assert", -1)
+			// find the revision/snap-id from the local provided
+			// assertion and set info accordingly. no need to
+			// actually check the assertion because we do that
+			// anyway for all snaps that have a snapID/revision.
+			f, err := os.Open(assertFile)
```

Comment:
I think the comment should also clarify that we will fetch the assertion again from the store, this is not making the process offline-friendly, just anchoring it